test/inductor/test_torchinductor.py

```diff
@@ -5305,6 +5305,18 @@ def fn(a, b):
                     kernel_profile_events.append(e.name)
             assert len(kernel_profile_events) > 0
 
+        def test_input_is_inplace_view(self):
+            @torch._dynamo.optimize("inductor")
+            def fn(a):
+                unsqueeze_ = torch.ops.aten.unsqueeze_.default(a, 0)
+                return unsqueeze_
+
+            args = [
+                ((1, 1, 1, 12, 11, 3), (396, 396, 396, 33, 3, 1), torch.int64, "cpu")
+            ]
+            args = [rand_strided(sh, st, dt, dev) for (sh, st, dt, dev) in args]
+            fn(*args)
```

Comment:
This example will actually mutate the metadata of `args[0]` in-place.  We should assert that that mutation happened correctly.